src/hooks/Tokens.ts

```diff
@@ -57,9 +59,53 @@ export function useAllTokens(): { [address: string]: Token } {
   return useTokensFromMap(allTokens, true)
 }
 
+interface BridgeInfo {
+  [chainId: SupportedChainId | number | string]: {
+    tokenAddress: string
+    originBridgeAddress: string
+    destBridgeAddress: string
+  }
+}
 export function useUnsupportedTokens(): { [address: string]: Token } {
+  const { chainId } = useActiveWeb3React()
+  const listsByUrl = useAppSelector((state) => state.lists.byUrl)
   const unsupportedTokensMap = useUnsupportedTokenList()
-  return useTokensFromMap(unsupportedTokensMap, false)
+  const unsupportedTokens = useTokensFromMap(unsupportedTokensMap, false)
+
+  // checks the default L2 lists to see if `bridgeInfo` has an L1 address value that is unsupported
+  const L2InferredBlockedTokens: { [address: string]: Token } = useMemo(() => {
```

Comment:
nit: lowercase, ie l2InferredBlockedTokens

Reply:
I think a better typing (because it is used for expansion) would be `typeof unsupportedTokens`.